src/Workspaces/Core/Portable/Formatting/Engine/AbstractTriviaDataFactory.AbstractComplexTrivia.cs

```diff
@@ -41,54 +41,21 @@ protected abstract class AbstractComplexTrivia<TToken, TTrivia> : TriviaDataWith
                 this.Spaces = spaces;
             }
 
-            protected abstract TToken ConvertToken(SyntaxToken token);
-            protected abstract SyntaxTrivia ConvertTrivia(TTrivia trivia);
-
             protected abstract void ExtractLineAndSpace(string text, out int lines, out int spaces);
             protected abstract TriviaData CreateComplexTrivia(int line, int space);
             protected abstract TriviaData CreateComplexTrivia(int line, int space, int indentation);
-            protected abstract TriviaDataWithList<TTrivia> Format(FormattingContext context, ChainedFormattingRules formattingRules, int lines, int spaces, CancellationToken cancellationToken);
+            protected abstract TriviaDataWithList Format(FormattingContext context, ChainedFormattingRules formattingRules, int lines, int spaces, CancellationToken cancellationToken);
             protected abstract bool ContainsSkippedTokensOrText(TriviaList list);
 
-            public TToken Token1
-            {
-                get
-                {
-                    return ConvertToken(_token1);
-                }
-            }
+            public SyntaxToken Token1 => _token1;
```

Comment:
Couldn't you go further and make this `public SyntaxToken { get; }`?  Or is _token not assigned in the constructor?